xocto/ranges.py

```diff
@@ -1034,3 +1038,34 @@ def as_finite_datetime_periods(
         finite_periods += [FiniteDatetimeRange(period.start, period.end)]
 
     return finite_periods
+
+
+def iterate_over_months(
+    period: FiniteDatetimeRange, *, tz: datetime.tzinfo
+) -> Iterator[FiniteDatetimeRange]:
+    """
+    Generate a sequence of finite datetime ranges spanning months during the period.
+
+    Ranges span each whole month between the start and end times, inclusive.
+
+    ie: given a period of (15/01, 15/03) -> [
+        (15/01->01/02),
+        (01/02->01/03),
+        (01/03->15/03)
+    ]
+
+    Datetimes will be aligned to midnight as at the supplied timezone.
```

Comment:
:panda_face: do we gain anything from doing this midnight alignment? What happens if we drop it entirely?

Reply:
I think we probably don't gain anything from this. The boundaries of the ranges will all be relative to the start datetime, but it would be just as easy and more flexible if we left it to the caller to do the midnight alignment if it's needed.

Reply:
I've removed the midnight alignment in 768be2148b8c65b218bc0da540efcdfaba57bf45